### PR TITLE
feat(Toolbar): Redesign web toolbar

### PR DESCRIPTION
**Pull request by vp8x8**
menu

### CODE REVIEW

File: react/features/toolbox/functions.web.js [hunk 1]
```diff
@@ -4,6 +4,32 @@ import { hasAvailableDevices } from '../base/devices';
...
+ *
+ * @param {number} width - The width of the screen.
+ * @returns {Set} The button set.
+ */
+export function getToolbarAdditionalButtons(width: number): Set<string> {
+    if (width <= 500) {
+        if (width <= 390) {
+            if (width <= 332) {
+                if (width <= 224) {
+                    return new Set();
```
Comment (saghul) [inline react/features/toolbox/functions.web.js:18]: Not even 1 button?

File: react/features/toolbox/functions.web.js [hunk 2]
```diff
@@ -4,6 +4,32 @@ import { hasAvailableDevices } from '../base/devices';
...
 declare var interfaceConfig: Object;
 
+/**
+ * Returns a set of button names to be displayed in the toolbox, based on the screen width.
+ *
+ * @param {number} width - The width of the screen.
+ * @returns {Set} The button set.
+ */
+export function getToolbarAdditionalButtons(width: number): Set<string> {
+    if (width <= 500) {
```
Comment (quitrk) [inline react/features/toolbox/functions.web.js:14]: you'd get a nicer indentation if you start from the smallest width

File: react/features/toolbox/functions.web.js (hunk 1 above)
Comment (vp8x8) [reply react/features/toolbox/functions.web.js:18]: If this set is empty the `audio`, `video` & `hangup` will be displayed.  Also the rest of the buttons will show in the overflow menu.

File: react/features/toolbox/functions.web.js (hunk 2 above)
Comment (vp8x8) [reply react/features/toolbox/functions.web.js:14]: Indeed, but this is _somehow_ optimised for making less computations on larger screens (which are the main target)

File: css/_variables.scss [hunk 3]
```diff
@@ -4,7 +4,8 @@
  * Style variables
  */
 $baseFontFamily: -apple-system, BlinkMacSystemFont, 'open_sanslight', 'Helvetica Neue', Helvetica, Arial, sans-serif;
-$hangupColor: #bf2117;
+$hangupColor:#DD3849;
+$hangupHoverColor: #F25363;
```
Comment (muscat1) [inline css/_variables.scss:8]: If I'm not mistaken, we should be careful to also add these to the branding repo.

File: react/features/toolbox/functions.web.js [hunk 4]
```diff
@@ -4,6 +4,32 @@ import { hasAvailableDevices } from '../base/devices';
...
+        if (width <= 390) {
+            if (width <= 332) {
```
Comment (muscat1) [inline react/features/toolbox/functions.web.js:16]: Could we extract these values to some meaningfully named constants for a bit more clarity?

File: css/_toolbars.scss [hunk 5]
```diff
@@ -48,254 +45,216 @@
...
+    cursor: pointer;
+    display: flex;
+    font-size: 14px;
+    font-weight: 400;
+    height: 40px;
+    line-height: 24px;
+    padding: 8px 16px;
+    box-sizing: border-box;
+
+    &:hover {
```
Comment (muscat1) [inline css/_toolbars.scss:149]: Please also wrap this in a media query for mobile usability.

File: react/features/toolbox/components/web/Toolbox.js [hunk 6]
```diff
@@ -1047,14 +1017,36 @@ class Toolbox extends Component<Props, State> {
...
+                    text = { t('localRecording.dialogTitle') } />,
+            this._shouldShowButton('mute-everyone')
+                && <MuteEveryoneButton
+                    key = 'mute-everyone'
+                    showLabel = { true } />,
+            this._shouldShowButton('livestreaming')
+                && <LiveStreamButton
+                    key = 'livestreaming'
+                    showLabel = { true } />,
+            <hr
```
Comment (muscat1) [inline react/features/toolbox/components/web/Toolbox.js:1047]: If, for some weird reason, several of these buttons were disabled, wouldn't the separators be shown on top of one another? I was wondering, if it's not too complicated, if we had a way to discriminate whether we should show these or not. (Similarly to the profile one)

**Review by muscat1 — CHANGES_REQUESTED** [review]
This looks awesome!
Left a few minor comments, but the one more significant thing I've noticed is that the toolbar seems to not take into account whether the chat is open. I think this is due to the fact that you used `clientWidth` from the store rather than the state variable `windowWidth` that takes chat state into account, but this does raise a valid point, namely if we should recompute the `clientWidth` store value based on chat state, rather than performing computations on site. WDYT?

**Comment by vp8x8** [comment]
> This looks awesome!
> Left a few minor comments, but the one more significant thing I've noticed is that the toolbar seems to not take into account whether the chat is open. I think this is due to the fact that you used `clientWidth` from the store rather than the state variable `windowWidth` that takes chat state into account, but this does raise a valid point, namely if we should recompute the `clientWidth` store value based on chat state, rather than performing computations on site. WDYT?

Good catch. I used the state version of the width.

File: react/features/toolbox/functions.web.js (hunk 4 above)
Comment (vp8x8) [reply react/features/toolbox/functions.web.js]: Done.

File: css/_toolbars.scss (hunk 5 above)
Comment (vp8x8) [reply css/_toolbars.scss]: Done.

File: css/_variables.scss (hunk 3 above)
Comment (vp8x8) [reply css/_variables.scss:8]: I'll keep this in mind. Thanks!

File: react/features/toolbox/components/web/Toolbox.js (hunk 6 above)
Comment (vp8x8) [reply react/features/toolbox/components/web/Toolbox.js]: Fixed it.

**Comment by vp8x8** [comment]
Jenkins, test this please.

**Pull request merged**